src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -164,4 +165,45 @@ private static string GetEnvironmentVariable(string variableName)
             return null;
         }
     }
+
+    private static bool EvaluateOpenTelemetryReference(string loaderAssemblyLocation)
+    {
+        var appReferenceAssemblies = Assembly.GetEntryAssembly().GetReferencedAssemblies();
+        var index = Array.FindIndex(appReferenceAssemblies, assembly => assembly.Name == "OpenTelemetry");
+        bool throwException = false;
+
+        if (index != -1)
+        {
+            try
+            {
+                // Look up for type with an assembly name, will load the library.
+                // OpenTelemetry assembly load happens only if app has reference to the package.
+                var openTelemetryType = Type.GetType("OpenTelemetry.Sdk, OpenTelemetry");
+                var loadedOTelAssembly = Assembly.GetAssembly(openTelemetryType);
+                var loadedOTelFileVersionInfo = FileVersionInfo.GetVersionInfo(loadedOTelAssembly.Location);
```

Comment:
Is it really necessary to trigger the load of the SDK at this point? I mean, the `AssemblyName` already has the version information. Is this to try to avoid a case in which the entry assembly references one version but it actually (somehow) ends up loading another?